lstchain/scripts/lstchain_dl1_to_dl2.py

```diff
@@ -74,11 +75,107 @@
+    field lengths are not fixed)
+
+    Returns: training_az_deg, training_zd_deg arrays containing the azimuth and
+    zenith of the training nodes (in degrees)
+
+    """
+
+    training_zd_deg = []
+    training_az_deg = []
+
+    for dir in training_dirs:
+        c1 = dir.find('_theta_') + 7
+        c2 = dir.find('_az_', c1) + 4
+        c3 = dir.find('_', c2)
+        training_zd_deg.append(float(dir[c1:c2 - 4]))
+        training_az_deg.append(float(dir[c2:c3]))
+
+    training_zd_deg = np.array(training_zd_deg)
+    training_az_deg = np.array(training_az_deg)
+    # The order of the pointings is irrelevant
+
+    return training_az_deg, training_zd_deg
+
+
+def apply_to_file(filename, models_dict, output_dir, config, models_path):
 
     data = pd.read_hdf(filename, key=dl1_params_lstcam_key)
 
+    interpolate_energy = False
+    interpolate_gammaness = False
+    interpolate_direction = False
+    # Read in the settings for the interpolation of Random Forest predictions
+    # in cos(zd). If activated this avoids the jumps of performance produced
+    # by the discrete set of pointings in the RF training sample.
+    if 'random_forest_zd_interpolation' in config.keys():
+        zdinter = config['random_forest_zd_interpolation']
+        if 'interpolate_energy' in zdinter.keys():
+            interpolate_energy = zdinter['interpolate_energy']
+        if 'interpolate_gammaness' in zdinter.keys():
+            interpolate_gammaness = zdinter['interpolate_gammaness']
+        if 'interpolate_direction' in zdinter.keys():
+            interpolate_direction = zdinter['interpolate_direction']
+
+    interpolate_rf = {'energy_regression': interpolate_energy,
+                      'particle_classification': interpolate_gammaness,
+                      'disp': interpolate_direction
+                      }
+
+    dl1_training_dir = None
+    training_pointings = None
+    if True in interpolate_rf.values():
+        logger.warning('Cos(zenith) interpolation will be used in:')
+        if interpolate_energy:
+            logger.warning('   energy reconstruction Random Forest')
+        if interpolate_gammaness:
+            logger.warning('   g/h classification Random Forest')
+        if interpolate_direction:
+            logger.warning('   direction reconstruction Random Forest')
+
+        if 'random_forest_zd_interpolation' in config.keys():
+            zdinter = config['random_forest_zd_interpolation']
+            if 'dl1_training_dir' in zdinter.keys():
+                dl1_training_dir = zdinter['dl1_training_dir']
+
+        if dl1_training_dir is None:
+            # Build name of DL1 MC training files assuming standard pattern:
+            dummy = models_path.as_posix().replace('/data/models', '/data/mc/DL1')
+            dl1_training_dir = (Path(dummy[:dummy.rfind('/dec')] +
+                                     '/TrainingDataset/GammaDiffuse/' +
+                                     dummy[dummy.rfind('/dec') + 1:]))
```

Comment:
I think this should go to `lstchain.paths` module.

Reply:
No longer applies